x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/installed_integration_set.ts

```diff
@@ -33,10 +33,59 @@ interface PackageInfo extends InstalledPackageBasicInfo {
 export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
   const packageMap: PackageMap = new Map<string, PackageInfo>([]);
 
+  const addPackage = (fleetPackage: PackageListItem): void => {
+    if (fleetPackage.type !== 'integration') {
+      return;
+    }
+    if (fleetPackage.status !== 'installed') {
+      return;
+    }
+
+    const packageKey = `${fleetPackage.name}`;
+    const existingPackageInfo = packageMap.get(packageKey);
+
+    if (existingPackageInfo != null) {
+      return;
+    }
+
+    // Actual `installed_version` is buried in SO, root `version` is latest package version available
+    // SO should be available as PackageListItem should be InstalledRegistry<RegistrySearchResult>
+    const installedPackageVersion = fleetPackage.savedObject?.attributes?.install_version;
```

Comment:
If `fleetPackage.status === 'installed'`, `fleetPackage.savedObject?.attributes?.install_version` must be defined and probably we should assert it here using `invariant()`.

Reply:
Type narrowing works only within the `if (fleetPackage.status === 'installed') { ... }` block

Reply:
I didn't realize you had removed the `ts-expect-error` when we were debugging... 😅  The early return 'if not installed' above actually does work, so I was just able to remove the safe accessors as-is 🙂